src/Domain/Structure/Neighbors.cpp

```diff
@@ -26,6 +26,14 @@ Neighbors<VolumeDim, IdType>::Neighbors(std::unordered_set<IdType> ids,
          "Cannot use a default-constructed OrientationMap in Neighbors.");
 }
 
+template <size_t VolumeDim, typename IdType>
+Neighbors<VolumeDim, IdType>::Neighbors(const IdType& id,
+                                        OrientationMap<VolumeDim> orientation)
+    : ids_(std::unordered_set{id}), orientation_(std::move(orientation)) {
```

Comment:
I think you can just delegate to the unordered_set constructor here, no need to reimplement it